CMakeLists.txt

```diff
@@ -323,15 +324,36 @@ if(USE_CCACHE)
   endif()
 endif()
 
-# Build type configuration - default to Release if none is set
+# Build type configuration - default to Release if none is set unless
+# code coverage is in use, in which case override to Debug
 
-if(NOT CMAKE_BUILD_TYPE)
+if (OPENVDB_CODE_COVERAGE)
+  message(WARNING "Code coverage requires debug mode, setting CMAKE_BUILD_TYPE "
+    "to Debug.")
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING
+    "Code coverage requires debug mode." FORCE
+  )
+elseif(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release CACHE STRING
     "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE
   )
 endif()
 message(STATUS "CMake Build Type: ${CMAKE_BUILD_TYPE}")
 
+# Code coverage configuration
+
+if(OPENVDB_CODE_COVERAGE)
+  # set gcov compile and link flags
```

Comment:
Where does gcov actually get invoked?

Reply:
Nothing that's natively supported in our CMake (we use it in our CI only). This is mainly because there are a suite of tools that can process the .gcno files that are generated to turn the data into an html report (for example). It's left up to the user to process the data that's generated.

Reply:
Ah I meant where are the CI changes which use these options, are they still to be added?

Reply:
Ah, I understand what you're asking. It's in the master branch of the repo already, included as part of the Azure changes. It currently gets called but doesn't find any code coverage files because this change hasn't been included, so reports code coverage of 0%.